### PR TITLE
Python 3 compatibility (closes #24)

### LINKED ISSUE #24: Python 3 support
┆Issue is synchronized with this [JIRA Story](https://ucsc-cgl.atlassian.net/browse/DDL-13)
┆Project Name: dos-dss-lambda
┆Issue Number: DDL-13

### PR DESCRIPTION
**Pull request by natanlao**
As we're waiting for a test suite, I independently confirmed that these changes don't break anything. (That said, the `_scroll_id` and `search_after` key discrepancy still exists.)

### CODE REVIEW

**Comment by natanlao** [comment]
This PR implements Python 3 compatibility while maintaining Python 2 compatibility.

**Comment by mjkrause** [comment]
The sentence _As we're waiting for a test suite, I independently confirmed that these changes don't break anything._ is difficult to comprehend. 

I'm not sure what you mean by "waiting for...", and "independent ..." from what, and what changes?

File: app.py [hunk 1]
```diff
@@ -142,7 +143,7 @@ def convert_reference_json(reference_json, data_object):
     # u'url': [u'gs://topmed-irc-share/genomes/NWD145710.b38.irc.v1.cram',
     # u's3://nih-nhlbi-datacommons/NWD145710.b38.irc.v1.cram']}
     data_object['size'] = reference_json['size']
-    data_object['urls'] = map(lambda x: {'url': x}, reference_json['url'])
+    data_object['urls'] = [{'url': x} for x in reference_json['url']]
```
Comment (mjkrause) [inline app.py]: I like that you turned the lambdas into list comprehensions - those are also somewhat cryptic, but a breeze compared to lambdas (my personal opinion).

File: app.py [hunk 2]
```diff
@@ -222,19 +223,16 @@ def list_data_bundles():
...
+        # header of the response
+        next_page_token = urlparse.parse_qs(
+            urlparse.urlparse(
+                res.links['next']['url']).query)['search_after'][0]
     # And convert the fqid message into a DOS id and version
     response = {}
     response['next_page_token'] = next_page_token
     try:
-        response['data_bundles'] = map(dss_list_bundle_to_dos, res.json()['results'])
+        response['data_bundles'] = list(map(dss_list_bundle_to_dos, res.json()['results']))
```
Comment (mjkrause) [inline app.py]: Is the `list` command needed for Python 3? How do you ensure Python 2 compatibility? Or would that raise an exception?

**Comment by natanlao** [comment]
@mjkrause Apologies, let me clarify: I'm working on introducing a standardized test suite for all of our DOS lambdas (TLP-492). This is the same one I've discussed at stand up. 

Since dos-dss-lambda currently lacks tests, I had to manually verify that the lambda worked as intended despite the changes made in this PR.

**Review by mjkrause — CHANGES_REQUESTED** [review]

File: app.py (hunk 2 above)
Comment (mjkrause) [reply app.py]: Hi Natan, thanks for taking the time to answer my comments / questions. In know you mentioned your work and actions during standup, but I see commits and PRs more as a form of record keeping so we can revisit it later and know what was done and why.

**Review by mjkrause — APPROVED** [review]

**Comment by natanlao** [comment]
Thank you for the review, @mjkrause!

**Pull request merged**